### ISSUE TITLE
Crashed on iOS EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x0000000a00000000

### ISSUE THREAD

**Issue by AbdOoSaed** [RevenueCat/purchases-flutter#373]
- [x] I have updated Purchases SDK to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/RevenueCat/purchases-flutter/blob/main/CONTRIBUTING.md)
- [x] I have searched the [Community](https://community.revenuecat.com)
- [x] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [x] I have searched for [existing Github issues](https://github.com/RevenueCat/purchases-flutter/issues)

**Steps to Reproduce**
1. 3% of users affected.
2. The crash got reported to the Firebase console.
3. I haven't found a way to reproduce it.

```
Crashed: com.apple.main-thread
0  libsystem_platform.dylib       0xfc4 _platform_strlen + 4
1  CoreFoundation                 0x9d14 __CFStringAppendFormatCore + 8712
2  CoreFoundation                 0x2947c _CFStringCreateWithFormatAndArgumentsReturningMetadata + 176
3  libsystem_trace.dylib          0xc1d8 _os_log_impl_dynamic + 184
4  libsystem_trace.dylib          0xbdd4 _os_log_with_args_impl + 168
5  CoreFoundation                 0x68054 _CFLogvEx3 + 188
6  Foundation                     0x478a0 _NSLogv + 124
7  libswiftFoundation.dylib       0x20928 specialized withVaList<A>(_:_:) + 444
8  PurchasesCoreSwift             0x180d8 $s18PurchasesCoreSwift6LoggerC10logHandleryyAA8LogLevelO_SStcvpZfiyAF_SStcfU_ + 252
9  PurchasesCoreSwift             0x1877c $s18PurchasesCoreSwift6LoggerC3log5level6intent7messageyAA8LogLevelO_AA0I6IntentOSStFZTf4nnnd_n + 212
10 PurchasesCoreSwift             0x18688 $s18PurchasesCoreSwift6LoggerC5debugyySSFZToTm + 80
11 Purchases                      0x29514 -[RCStoreKitWrapper paymentQueue:removedTransactions:] + 620
12 libdispatch.dylib              0x1e68 _dispatch_call_block_and_release + 32
13 libdispatch.dylib              0x3a2c _dispatch_client_callout + 20
14 libdispatch.dylib              0x11f48 _dispatch_main_queue_drain + 928
15 libdispatch.dylib              0x11b98 _dispatch_main_queue_callback_4CF + 44
16 CoreFoundation                 0x522f0 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
17 CoreFoundation                 0xc1f4 __CFRunLoopRun + 2532
18 CoreFoundation                 0x1f6b8 CFRunLoopRunSpecific + 600
19 GraphicsServices               0x1374 GSEventRunModal + 164
20 UIKitCore                      0x513e88 -[UIApplication _run] + 1100
21 UIKitCore                      0x2955ec UIApplicationMain + 364
22 Runner                         0x77a0 main + 9 (AppDelegate.swift:9)
23 ???                            0x100359ce4 (Missing)
```

**Comment by AbdOoSaed** [comment]
last version
![ss](https://user-images.githubusercontent.com/33700292/170814980-0bdb0e72-e931-4f16-a735-d658057aa4bb.JPG)

**Comment by AbdOoSaed** [comment]
Any update

**Comment by vegaro** [comment]
Hi @AbdOoSaed , sorry we didn't reply earlier. I looked into it and we think this has been solved with a more recent version of the iOS dependency we use in our Flutter plugin. We are currently working on making all the required changes to update to this newer iOS version and we'll release that as purchases-flutter 4.0.0. We already have a WIP PR for it, in case you want to keep track of the work https://github.com/RevenueCat/purchases-flutter/pull/354
